File: JavaScript/Maths/nPuissanceN.js
/**
 * @description Fonction pour verifier si un nombre est n-puissance de n
 * @param {number} n - Nombre a verifier
 * @param {number} int - Entier positif
 * @returns {boolean} True si le nombre est n-puissance de n, false sinon
 */
const nPuissanceN = (n, int) => int ** int === n;

// Test
console.log(nPuissanceN(3125, 5)); // true, 5^5 = 3125

File: Python/Maths/n_puissance_n.py
def n_puissance_n(n: int, p: int) -> bool:
    """
    Description:
        Fonction pour verifier si un nombre est n-puissance de n

    Paramètres:
        n: {int} -- Nombre à verifier
        p: {int} -- Nombre de puissance

    Retourne:
        {bool} -- True si le nombre est n-puissance de n, False sinon
    
    Exemple:
        >>> n_puissance_n(3125, 5)
        True
    """
    return n == p ** p